cmd/konflux/main.go

```
@@ -311,36 +314,48 @@ func generateTekton(application k.Application, target string) error {
-			if err := generateFileFromTemplate("component-push.yaml", component, filepath.Join(target, fmt.Sprintf("%s-%s-%s-push.yaml", hyphenize(basename(application.Repository)), hyphenize(application.Version), c))); err != nil {
+			if err := generateFileFromTemplate("component-push.yaml", c, filepath.Join(target, fmt.Sprintf("%s-%s-%s-push.yaml", hyphenize(basename(application.Repository)), hyphenize(application.Version), c.Name))); err != nil {
 				return err
 			}
 		default:
-			if err := generateFileFromTemplate("component-pull-request.yaml", component, filepath.Join(target, fmt.Sprintf("%s-%s-%s-pull-request.yaml", hyphenize(basename(application.Repository)), hyphenize(application.Version), c))); err != nil {
+			if err := generateFileFromTemplate("component-pull-request.yaml", c, filepath.Join(target, fmt.Sprintf("%s-%s-%s-pull-request.yaml", hyphenize(basename(application.Repository)), hyphenize(application.Version), c.Name))); err != nil {
 				return err
 			}
-			if err := generateFileFromTemplate("component-push.yaml", component, filepath.Join(target, fmt.Sprintf("%s-%s-%s-push.yaml", hyphenize(basename(application.Repository)), hyphenize(application.Version), c))); err != nil {
+			if err := generateFileFromTemplate("component-push.yaml", c, filepath.Join(target, fmt.Sprintf("%s-%s-%s-push.yaml", hyphenize(basename(application.Repository)), hyphenize(application.Version), c.Name))); err != nil {
 				return err
 			}
 		}
 	}
 	return nil
 }
 
+// This function can modified in future if we want to override the fields at component level.
+func updateComponent(application k.Application, c *k.Component) {
+	c.Application = application.Name
+	c.Repository = application.Repository
+	c.Branch = application.Branch
+	c.Version = application.Version
+	c.Platforms = application.Platforms
+	if c.Tekton == (k.Tekton{}) {
+		c.Tekton = application.Tekton
+	}
+	if c.Dockerfile == "" {
+		c.Dockerfile, _ = eval(".konflux/dockerfiles/{{.Name}}.Dockerfile", c)
```

why ignoring the error ?

---

Fixed